src/Auth/Identifiable.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Auth;
+
+interface Identifiable
```

Comment:
I think that, if we separate the two concerns of "resolving an identifiable" and "managing the session", we'll probably have to refactor this interface as well.